src/overlap_check/test/overlap_check_test.hpp

```diff
@@ -0,0 +1,16 @@
+
+#include "gtest/gtest.h"
+
+using namespace moab;
+
+class OverlapTest : public::testing::Test {
+ protected:
+  virtual void SetUp();
+  virtual void TearDown();
```

Comment:
Both of these functions should be marked with `override`. eg.:
```c++
    virtual void Setup() override;
    virtual void TearDown() override;
```
This is more of a discipline that is good to follow as it makes it more likely that mistakes with virtual/overridden functions will result in compiler errors, making them easier to address.